apps/glusterfs/health_cache.go

```diff
@@ -154,14 +154,15 @@ func (hc *NodeHealthCache) toProbe() ([]*NodeHealthStatus, error) {
 			if err != nil {
 				return err
 			}
-			// Ignore if the node is not online
-			if !node.isOnline() {
-				continue
-			}
 			nhs := &NodeHealthStatus{
 				NodeId: nodeId,
 				Host:   node.Info.Hostnames.Manage[0],
 			}
+			// Ignore if the node is not online
+			if !node.isOnline() {
+				continue
+			}
+			nhs.Up = true
```

Comment:
I don't understand this change. What was wrong with the previous behavior? How does this change improve it? Because this is a subtle thing this is the kind of change I'd prefer to see in a separate commit with it's own commit message explaining the change (assuming it is still needed).

Reply:
i have added  it to todo list to make seperate commit,

currently, we  are saving only online node details  in the slice, for the health
implementation, we need  all nodes  status  irrespective of its state to decide whether
the cluster state is healthy or not.

Reply:
Ok, makes sense. I do wonder if it would be beneficial to keeping two slices, one for up/healthy nodes and one for down/unhealthy nodes. That way when one needs only the list of healthy nodes code only needs to consult one slice w/o needing any additional conditionals.